### DISCUSSION TITLE
Building docs for a Python 3.6 package

### DISCUSSION THREAD

**Discussion by in03** [mkdocstrings/mkdocstrings#402] [Q&A] [answered]
Hi all,

Potentially a basic question here. I'm using Python 3.6 out of necessity to support a proprietary Python API that doesn't work for later versions of Python. I've been following the [recipe](https://mkdocstrings.github.io/recipes/), but realised the majority of the necessary plugins (as well as mkdocstrings) requires 3.7.

I'm building the docs with GitHub actions for my repo, so I can easily run this as the required 3.7. For development I'm using Poetry, which has been great to manage dependencies and virtual environments, but has been a pain to manage when working with multiple environments for one project on Windows. I did get it working, but it does not allow using a virtual environment newer than the maximum version specified in `pyproject.toml`. This seems to be standard fare with dependency management tools.

I'd like to have this all a nice and smooth CI/CD process but not sure of an easy solution. I'm aware it's considered bad practice to install python libraries globally. Should I perhaps use another tool to manage a separate venv for docs? One that ignores `pyproject.toml`?

Apologies if this question is off-topic.
Any suggestions would be appreciated.

**Comment by pawamoy** [comment, marked as answer]
So, to try and summarize: you have `python <= 3.6` as a requirement, and would like to benefit from Poetry. As you said, Poetry won't be happy and won't resolve deps as mkdocstrings and plugins are `python >= 3.7`. Apart from using a venv manually, I don't see a solution for Poetry. With PDM you could maybe use the `overrides` option: https://pdm.fming.dev/usage/dependency/#solve-the-locking-failure

**Reply by in03** [reply]
Thanks Pawamoy, that makes a lot of sense :) Apologies for the delay in marking as answer

**Reply by pawamoy** [reply]
No problem at all, thank you for coming back and marking the answer :slightly_smiling_face:
